core/state/snapshot/account.go

```diff
@@ -52,3 +52,30 @@ func AccountRLP(nonce uint64, balance *big.Int, root common.Hash, codehash []byt
 	}
 	return data
 }
+
+// conversionAccount is used for converting between full and slim format. When
+// doing this, we can consider 'balance' as a byte array, as it has already
+// been converted from big.Int into an rlp-byteslice.
+type conversionAccount struct {
+	Nonce    uint64
+	Balance  []byte
+	Root     []byte
+	CodeHash []byte
+}
+
+// SlimToFull converts data on the 'slim RLP' format into the full RLP-format
+func SlimToFull(data []byte) []byte {
```

Comment:
Please return an error if rlp.Decode fails. We'll need it when parsing data from the snap protocol.

Reply:
I think I'd move this and `conversionAccount` into `conversion.go` until it's clear where it goes. Don't want to litter the package. I'm unsure this is a method we want to keep long term, seems weird to do three ops in on (decode, convert, recode).